### PR TITLE
dts: Add base binding for interrupt-controller, clock, gpio, gpio-nexus, and fixup pwm base

### PR DESCRIPTION
**Pull request by galak**
Add base bindings "classes" for interrupt-controller, clock, gpio, and gpio-nexus.  Update the pwm base to match style of the new base bindings and cleanup some additional properties in pwm.

### CODE REVIEW

File: dts/bindings/interrupt-controller/intc.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#
+# Copyright (c) 2019, Linaro Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+title: Interrupt Controller Base Structure
+
+description: >
+    This binding gives the base structures for all Interrupt Controllers devices
```
Comment (ioannisg) [inline dts/bindings/interrupt-controller/intc.yaml]: base structure
Controller devices

**Review by ioannisg — COMMENTED** [review]
Looks ok to me

File: dts/bindings/interrupt-controller/intc.yaml (hunk 1 above)
Comment (galak) [reply dts/bindings/interrupt-controller/intc.yaml]: Fixed.

**Review by ulfalizer — APPROVED** [review]
Looks fine to me.

**Comment by ulfalizer** [comment]
Maybe `gpio-controller.yaml` would be clearer than `gpio.yaml`, etc.

**Comment by galak** [comment]
> Maybe `gpio-controller.yaml` would be clearer than `gpio.yaml`, etc.

Changed gpio and interrupt to be gpio-controller.yaml & interrupt-controller.yaml

**Pull request merged**